sdk/core/azure-core/tests/test_connection_string_parsing.py

```diff
@@ -0,0 +1,182 @@
+        parse_result = parse_connection_string(conn_str, True)
+        assert parse_result["enDpoInt"] == 'XXXXENDPOINTXXXX'
+        assert parse_result["Endpoint"] == 'XXXXENDPOINT2XXXX'
+
+    def test_error_with_duplicate_case_insensitive_keys_for_insensitive_conn_str(self, **kwargs):
+        conn_str = 'endPoinT=XXXXENDPOINTXXXX;eNdpOint=XXXXENDPOINT2XXXX;sharedaccesskeyname=XXXXPOLICYXXXX;SHAREDACCESSKEY=THISISATESTKEYXXXXXXXXXXXXXXXXXXXXXXXXXXXX='
+        with pytest.raises(ValueError) as e:
+            parse_result = parse_connection_string(conn_str, False)
+        assert str(e.value) == "Duplicate key in connection string: endpoint"
+    
+    def test_error_with_malformed_conn_str(self):
+        for conn_str in ["", "foobar", "foo;bar;baz", ";", "foo=;bar=;", "=", "=;=="]:
+            with pytest.raises(ValueError) as e:
+                parse_result = parse_connection_string(conn_str)
+            self.assertEqual(str(e.value), "Connection string is either blank or malformed.")
+
+    def test_case_insensitive_clear_method(self):
+        conn_str = 'enDpoiNT=XXXXENDPOINTXXXX;sharedaccesskeyname=XXXXPOLICYXXXX;SHAREDACCESSKEY=THISISATESTKEYXXXXXXXXXXXXXXXXXXXXXXXXXXXX='
+        parse_result = parse_connection_string(conn_str, False)
+        parse_result.clear()
+        assert len(parse_result) == 0
+
+    def test_case_insensitive_copy_method(self):
+        conn_str = 'enDpoiNT=XXXXENDPOINTXXXX;sharedaccesskeyname=XXXXPOLICYXXXX;SHAREDACCESSKEY=THISISATESTKEYXXXXXXXXXXXXXXXXXXXXXXXXXXXX='
+        parse_result = parse_connection_string(conn_str, False)
+        copied = parse_result.copy()
+        assert copied == parse_result
+    
+    def test_case_insensitive_get_method(self):
+        conn_str = 'Endpoint=XXXXENDPOINTXXXX;SharedAccessKeyName=XXXXPOLICYXXXX;SharedAccessKey=THISISATESTKEYXXXXXXXXXXXXXXXXXXXXXXXXXXXX='
+        parse_result = parse_connection_string(conn_str, False)
+        assert parse_result.get("Endpoint") == 'XXXXENDPOINTXXXX'
+        assert parse_result.get("endPoint") == 'XXXXENDPOINTXXXX'
+        assert parse_result.get("SharedAccessKeyName") == 'XXXXPOLICYXXXX'
+        assert parse_result.get("sharedAccESSkEynAME") == 'XXXXPOLICYXXXX'
+        assert parse_result.get("SharedAccessKey") == 'THISISATESTKEYXXXXXXXXXXXXXXXXXXXXXXXXXXXX='
+        assert parse_result.get("sharedaccesskey") == 'THISISATESTKEYXXXXXXXXXXXXXXXXXXXXXXXXXXXX='
+        assert parse_result.get("accesskey") is None
+        assert parse_result.get("accesskey", "XXothertestkeyXX=") == "XXothertestkeyXX="
+
+    def test_case_insensitive_keys_method(self):
+        conn_str = 'enDpoiNT=XXXXENDPOINTXXXX;sharedaccesskeyname=XXXXPOLICYXXXX;SHAREDACCESSKEY=THISISATESTKEYXXXXXXXXXXXXXXXXXXXXXXXXXXXX='
+        parse_result = parse_connection_string(conn_str, False)
+        keys = parse_result.keys()
+        assert len(keys) == 3
+        assert "enDpoiNT" in keys
+        if sys.version_info >= (3, 5):
```

Comment:
The behavior of the `.keys()` method differs in Python 2.7 vs. 3.5+ for CaseInsensitiveDict. Is this concerning behavior?